apps/rpc/src/modules/event/attendance-service.ts

```diff
@@ -0,0 +1,795 @@
+      // from registration start.
+      const mergeablePools = attendance.pools.filter(isMergeable)
+      const pendingPools = attendance.pools.filter((pool) => !isMergeable(pool))
+      // Depending on if there is zero or one pools, we either update the matching pool, or do nothing
+      if (mergeablePools.length <= 1) {
+        if (mergeablePools.length === 1) {
+          const pool = mergeablePools.at(0)
+          invariant(pool !== undefined)
+          await attendanceRepository.updateAttendancePoolById(handle, pool.id, {
+            ...pool,
+            title: data.title,
+          })
+        }
+        return
+      }
+
+      // We compute the next properties by summing up for all the pools. The next pool should not have a merge delay
+      // since a potential pending pool should be merged into it.
+      const defaultMergePool = {
+        title: data.title,
+        mergeDelayHours: null,
+        capacity: 0,
+        yearCriteria: [] as number[],
+      } satisfies AttendancePoolWrite
+      const input = mergeablePools.reduce((acc, curr) => {
+        return {
+          title: acc.title,
+          mergeDelayHours: acc.mergeDelayHours,
+          capacity: acc.capacity + curr.capacity,
+          yearCriteria: acc.yearCriteria.concat(...curr.yearCriteria),
+        }
+      }, defaultMergePool)
+      validateAttendancePoolWrite(input)
+      validateAttendancePoolDisjunction(input.yearCriteria, pendingPools)
+      const pool = await attendanceRepository.createAttendancePool(handle, attendanceId, input)
+      const mergeablePoolIds = mergeablePools.map((pool) => pool.id)
+      await attendanceRepository.updateAttendeeAttendancePoolIdByAttendancePoolIds(handle, mergeablePoolIds, pool.id)
+      await attendanceRepository.deleteAttendancePoolsByIds(handle, mergeablePoolIds)
+    },
+  }
+}
+
+function isSelectionEqual(left: AttendanceSelection, right: AttendanceSelection): boolean {
+  const isWeaklyEqual = left.id === right.id && left.name === right.name && left.options.length === right.options.length
+  const isOptionSetEqual = left.options.every((lOption) => {
+    const rOption = right.options.find((rOption) => rOption.id === lOption.id)
+    return rOption?.name === lOption.name
+  })
+  return isWeaklyEqual && isOptionSetEqual
+}
+
+function validateAttendanceWrite(data: AttendanceWrite) {
+  if (isBefore(data.registerEnd, data.registerStart)) {
+    throw new AttendanceValidationError("Cannot specify registration end before start")
+  }
+  if (Math.abs(differenceInHours(data.registerStart, data.registerEnd)) < 1) {
+    throw new AttendanceValidationError("Registration time must be at least one hour long")
+  }
+}
+
+function validateAttendancePoolWrite(data: AttendancePoolWrite) {
+  if (data.mergeDelayHours !== null && (data.mergeDelayHours < 0 || data.mergeDelayHours > 48)) {
+    throw new AttendanceValidationError("Merge delay for pool must be between 0 and 48 hours")
```

Comment:
Why? Can't we just make sure that `registerStart + mergeDelayHours <= deregistrationDeadline` or something?